frontend/src/scenes/sessions/sessionsTableLogic.ts

```diff
@@ -176,8 +186,17 @@ export const sessionsTableLogic = kea<sessionsTableLogicType<SessionRecordingId>
                 ),
         ],
         expandedRowKeys: [
-            (selectors) => [selectors.sessions],
-            (sessions: SessionType[]): string[] => sessions?.map((s) => s.global_session_id) || [],
+            (selectors) => [selectors.sessions, selectors.rowExpandState],
+            (sessions: SessionType[], rowExpandState: ExpandState): string[] | boolean => {
+                if (rowExpandState === ExpandState.collapsed) {
+                    return []
+                }
+                if (rowExpandState === ExpandState.manual) {
+                    return false
+                }
+                // expand all
+                return sessions?.map((s) => s.global_session_id) || []
```

Comment:
Is this the case when `ExpandState.expanded`? Code could also be cleaner here to indicate that. 

Example:

```
switch(rowExpandState) {
  case ExpandState.collapsed: 
    return []
  case ExpandState.collapsed: 
    return something
  case ExpandState.expanded:
    return something
}
```

This has the benefit of erroring when a new expand state is added

Reply:
Is the `?` and default needed here? The type above `sessions: SessionType[]` indicates it's always an array. If that's not true, we should fix the type first.